lib/scidata/yelp_polarity_reviews.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Scidata.YelpPolarityReviews do
+  @moduledoc """
+  Module for downloading the [Yelp Reviews dataset](https://www.yelp.com/dataset).
+  """
+
+  @base_url "https://s3.amazonaws.com/fast-ai-nlp/"
+
+  @dataset_file "yelp_review_polarity_csv.tgz"
+
+  alias Scidata.Utils
+
+  @doc """
+  Downloads the Yelp Polarity Reviews training dataset or fetches it locally.
+  """
+  @spec download() :: %{review: [binary(), ...], sentiment: 1 | 0}
+  def download(), do: download_dataset(:train)
+
+  @doc """
+  Downloads the Yelp Polarity Reviews test dataset or fetches it locally.
+  """
+  @spec download_test() :: %{
+          review: [binary(), ...],
+          sentiment: 1 | 0
+        }
+  def download_test(), do: download_dataset(:test)
+
+  defp download_dataset(dataset_type) do
+    files = Utils.get!(@base_url <> @dataset_file).body
+    regex = ~r"#{dataset_type}"
+
+    records =
+      for {fname, contents} <- files,
+          List.to_string(fname) =~ regex,
+          reduce: [[]] do
+        _ -> parse_csv(contents)
+      end
+
+    %{
+      review: records |> Enum.map(&List.last(&1)),
+      sentiment: get_rating(records)
+    }
+  end
+
+  defp parse_csv(content) do
+    content
+    |> StringIO.open()
+    |> elem(1)
+    |> IO.binstream(:line)
+    |> CSV.decode!()
+    |> Enum.to_list()
+  end
+
+  defp get_rating(records) do
+    records
+    |> Enum.map(fn x ->
+      x
+      |> List.first()
+      |> case do
+        "1" -> 0
+        "2" -> 1
+      end
+    end)
```

Comment:
```suggestion
    Enum.map(records, fn
      ["1" | _] -> 0
      ["2" | _] -> 1
    end)
```

Reply:
This is indeed much so much cleaner! I've made this change.